nexus/tests/integration_tests/disks.rs

```diff
@@ -822,6 +829,126 @@ async fn test_disk_reject_total_size_not_divisible_by_min_disk_size(
+    create_instance_with_disk(client).await;
+
+    // NOTE: This is a little gnarly, but we sleep long enough to allow some
+    // metrics to be populated. Noted that this relies on the producer server
+    // within the simulated disk using an interval smaller than a second.
+    tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+
+    for metric in &ALL_METRICS {
+        let measurements: ResultsPage<Measurement> =
+            objects_list_page_authz(client, &metric_url(metric)).await;
+        assert!(!measurements.items.is_empty());
+        for item in &measurements.items {
+            let cumulative = match item.datum() {
+                Datum::CumulativeI64(c) => c,
+                _ => panic!("Unexpected datum type {:?}", item.datum()),
+            };
+            assert!(cumulative.start_time() <= item.timestamp());
+        }
+    }
+}
+
+#[nexus_test]
+async fn test_disk_metrics_paginated(cptestctx: &ControlPlaneTestContext) {
+    let client = &cptestctx.external_client;
+    DiskTest::new(&cptestctx).await;
+    create_org_and_project(client).await;
+    create_disk(&client, ORG_NAME, PROJECT_NAME, DISK_NAME).await;
+    create_instance_with_disk(client).await;
+
+    // NOTE: This is a little gnarly, but we sleep long enough to allow some
+    // metrics to be populated. Noted that this relies on the producer server
+    // within the simulated disk using an interval smaller than a second.
+    tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+
+    for metric in &ALL_METRICS {
+        let collection_url =
+            format!("{}/{DISK_NAME}/metrics/{metric}", get_disks_url(),);
+        let initial_params = format!(
+            "start_time={:?}&end_time={:?}",
+            Utc::now() - chrono::Duration::seconds(5),
+            Utc::now(),
+        );
+        let measurements_paginated: Collection<Measurement> =
+            NexusRequest::iter_collection_authn(
+                client,
+                &collection_url,
+                &initial_params,
+                Some(10),
+            )
+            .await
+            .expect("failed to iterate over metrics");
+        assert!(!measurements_paginated.all_items.is_empty());
+
+        let mut last_timestamp = None;
+        let mut last_value = None;
+        for item in &measurements_paginated.all_items {
+            let cumulative = match item.datum() {
+                Datum::CumulativeI64(c) => c,
+                _ => panic!("Unexpected datum type {:?}", item.datum()),
+            };
+            assert!(cumulative.start_time() <= item.timestamp());
+
+            // Validate that the timestamps increase.
+            if let Some(last_ts) = last_timestamp {
+                assert!(last_ts < item.timestamp());
+            }
+            // Validate that the values increase.
+            if let Some(last_value) = last_value {
+                assert!(last_value < cumulative.value());
```

Comment:
It seems safer to make this an inequality check, `<=`. Cumulative values are non-decreasing, not necessarily increasing.

Reply:
Done.